app/views/home/landing.html.erb

```diff
@@ -22,121 +22,173 @@
           </div>
         </div>
       </div>
-      <section class="queryOptionsContainer" id="default">
-        <% @capabilities.to_a.each do |c| %>
-          <div data-type="capabilities" class="pill white"><%= c.name %></div>
-        <% end %>
-        <% @materials.to_a.each do |m| %>
-          <div data-type="materials" class="pill white"><%= m.name %></div>
-        <% end %>
-      </section>
-      <section class="queryOptionsContainer" id="results" style="display:none;"> </section>
+      <section class="queryOptionsContainer" id="results"> </section>
       <footer class="homeFooter">
         <p>
           o consulta el <%= link_to equipment_index_path do %>
-            <%= content_tag :strong, "cat&aacute;logo de equipos".html_safe, :class => "bold" %>
+            <%= content_tag :strong, "catálogo de equipos".html_safe, :class => "bold" %>
           <% end %>
         </p>
       </footer>
       <%= render partial: 'shared/search' %>
     </div>
     <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
     <script>
-      const $pillBox = $('.queryOptionsContainer#results');
-      const $defaultPillBox = $('.queryOptionsContainer#default');
-
-      window.capabilities = <%= raw(@capabilities.as_json(only: [:id, :name]).to_json) %>;
-      window.materials = <%= raw(@materials.as_json(only: [:id, :name]).to_json) %>;
-
+      // Container for pills returned from a query
+      const $pillBox = $('.queryOptionsContainer');
+      // Input box for capacities
       const $queryCap = $("#query-capacidades");
+      // Input box for materials
       const $queryMat = $("#query-materiales");
-
-      function draw(results, type) {
-        const resultDivs = results.map(r => {
+      // Container for selected capacity
+      const $selectedCap = $(".tagBox#capabilities");
+      // Container for selected material
+      const $selectedMat = $(".tagBox#materials");
+
+      // Retuns the HTML to render pills
+      function getHTMLForPills(results, type) {
+        return results.map(r => {
           return `<div data-id="${r.id}" data-type="${type}" class="pill white">${r.name}</div>`;
         });
-        $pillBox.html(resultDivs);
       }
 
-      const processQueryOnCollection = function(collection) {
-        return function() {
-          const query = $(this).val().trim().toLowerCase();
-          if(query == "") {
-            console.log(query);
-            $pillBox.hide();
-            $defaultPillBox.show();
-            return false;
-          }
-
-          const results = window[collection].filter(c => c.name.toLowerCase().indexOf(query) > -1);
-          $defaultPillBox.hide();
-          $pillBox.show();
-          draw(results, collection);
-        };
-      };
+      // Creates the pills for the items in results (array)
+      function draw(results, type) {
+        let pillsHTML = getHTMLForPills(results, type);
+        $pillBox.html(pillsHTML);
+      }
```

Comment:
You're changing most of the js, would it be better to make a separate js file?

Reply:
Done